datastructures/hashmap/withProbing.go

```diff
@@ -0,0 +1,136 @@
+package hashmap
+
+import (
+	"fmt"
+	"math"
+)
+
+const (
+	goldenRatio     float64 = 1.618033988749895
+	radix                   = uint64('~' - ' ')
+	loadFactor      float64 = 0.667
+	initMValue              = 1
+	deletedEntryKey         = "U+0000"
+)
+
+type entry struct {
+	key   interface{}
+	value interface{}
+}
+type table struct {
```

Comment:
Please capitalise your main structure that you will want to be used in other package level code 😄

```suggestion
type Table struct {
```

Reply:
This would also require you to make its fields that someone may want to use outside for example, `size` to `Size` would be accessible in outside scope.

Reply:
I'm providing Hash table interface to the user, so there is no point in exporting Table or Size, 
for Size , I will add Len - method to the interface

Reply:
Ah I see. I didn't see it as an example implementation.

Reply:
Should I export those fields and struct? and remove the Hash Table interface ?

Reply:
Up to you actually, I think I don't see how it can be used more generally than what you've already implemented. From what I understand you are using `HashTable` interface as a `return` type in the function `NewHashtable` which is essentially sending the `newTable` from the other file. So it is redundant in some sense, but at the same time this is indeed the proper way of doing it - because then people can just use the `HashTable` as an interface in functions to make it more general.

These are my personal opinions on your implementation so I think you should wait for the maintainers to see your code before making any changes.

Reply:
Consider having an `Example` (if you are familiar with `Test Examples` in go - [here](https://blog.golang.org/examples) is the link if not) implementation so that people can see how they can implement something general that would work with that interface.

Again, I would say that it is worth waiting for the maintainers to review your code before making these changes.